var/spack/repos/builtin/packages/py-datalad/package.py

```diff
@@ -99,14 +112,21 @@ class PyDatalad(PythonPackage):
         # downloaders-extra
```

Comment:
It's kinda annoying to maintain this list twice. Should we just get rid of `+full`?

Reply:
Yes it is really annoying, unfortunately getting rid of `+full` is not an option (yet) because it is needed to build `py-datalad-hirni`. I saw that in the main branch of `py-datalad-hirni` they only have a dependency on `py-datalad` without the `+full` but there is no release yet.

Reply:
Could we replace:
```python
depends_on("py-datalad+full@0.14.0:", type=("build", "run"))
```
with:
```python
depends_on("py-datalad+downloaders-extra+misc+tests+metadata-extra+duecredit@0.14.0:", type=("build", "run"))
```
?

Reply:
You mean and than get rid of full `+full` in `py-datalad-hirni? 
We can either do that or we could add conflicts to `py-datalad`
```
    conflicts("~downloaders-extra", when="+full")
    conflicts("~misc", when="+full")
    conflicts("~tests", when="+full")
    conflicts("~metadata-extra", when="+full")
    conflicts("~duecredit", when="+full")
```
Then the concretizer does the work for us.

Reply:
Yep, I'm fine with either one of those

Reply:
Ok, I have added the conflict approach.